### PR TITLE
Migrate to webpack@2

### PR DESCRIPTION
**Pull request by 7rulnik**
Okay, I did it (#200).

But stuck on Sinon error:
https://monosnap.com/file/vCQv2CPHLt7JsxuBIcPpG9ULRSK2O2

### CODE REVIEW

**Comment by mxstbr** [comment]
Hmm, I don't think this is everything that needs to be done – IIRC we need to switch to `System.import` for our code splitting routes and the reducer, right?

**Comment by 7rulnik** [comment]
Oh, I forgot about this.
Will do it tomorrow

**Comment by 7rulnik** [comment]
Just add code splitting.
Something else to do?

File: package.json [hunk 1]
```diff
@@ -125,9 +125,6 @@
       "es2015-native-modules",
```
Comment (7rulnik) [inline package.json]: foo

**Comment by 7rulnik** [comment]
Something else?

**Comment by mxstbr** [comment]
`git rebase -i HEAD~3`, then squash the "test commit" and the reversion into the "Replace babel preset" commit. Should look something like this:

```
pick   50b0574 Replace Babel preset
squash 0021ec0 test commit
squash 18b1395 Revert "test commit"
```

**Comment by mxstbr** [comment]
This looks good otherwise, could you `git merge v3.0.0` to include the newest changes from the `v3.0.0` branch? Thanks!

**Comment by 7rulnik** [comment]
@mxstbr, are we ready to merge?

**Comment by mxstbr** [comment]
Still getting some errors and conflicts here...

File: internals/templates/routes.js [hunk 2]
```diff
@@ -4,22 +4,39 @@
...
 export default function createRoutes(store) { // eslint-disable-line
   return [
     {
       path: '/',
-      getComponent: function get(location, cb) {
-        require.ensure([], (require) => {
-          cb(null, require('HomePage').default);
-        }, 'HomePage');
+      getComponent(location, cb) {
+        System.import('HomePage/reducer')
```
Comment (fubhy) [inline internals/templates/routes.js]: You are using the module loader twice here. So you are generating two promises, one for the reducer, one for the page. But you are calling the callback directly after the second one is resolved. So if the second one resolves faster, the route is loaded before the reducer is attached (unless I am misunderstanding this). Conclusion: You should probably use Promise.all() here.

Comment (7rulnik) [inline internals/templates/routes.js]: Yes, you're right. Thanks!

**Comment by 7rulnik** [comment]
There is error with Sinon and fetch and I have no idea how fix it :disappointed:

File: internals/templates/routes.js [hunk 3]
```diff
@@ -4,22 +4,39 @@
...
-          cb(null, require('HomePage').default);
-        }, 'HomePage');
+      getComponent(location, cb) {
+        Promise.all([
+          System.import('HomePage/reducer'),
+          System.import('HomePage'),
+        ]).then(modules => {
+          loadReducer(store, 'home')(modules[0]);
+          loadModule(cb)(modules[1]);
+        }).catch(errorLoading);
```
Comment (mxstbr) [inline internals/templates/routes.js]: The Homepage doesn't have a reducer associated with it in the cleaned state, so the async reducer loading here is not necessary!

File: package.json [hunk 4]
```diff
@@ -85,12 +86,12 @@
     "rimraf": "^2.5.2",
     "sanitize.css": "^3.3.0",
     "shelljs": "^0.6.0",
-    "sinon": "^1.17.3",
+    "sinon": "^2.0.0-pre",
```
Comment (mxstbr) [inline package.json]: The sinon error is almost definitely due to the upgrade to a pre-release version of sinon 2.0. (ref https://github.com/sinonjs/sinon/pull/932)

Do we need to update it?

Comment (mxstbr) [inline package.json]: Nevermind, this doesn't work when using sinon@1.17.3 either, I'd still opt to not use a pre-release version of it. This problem is probably Karma and/or webpack related, I'll dig further!

**Comment by jurekbarth** [comment]
FYI it works fine for me now :) All 63 tests passed

**Comment by mxstbr** [comment]
@jurekbarth did you do a clean install run?

**Comment by jurekbarth** [comment]
@mxstbr yes :)

**Comment by mxstbr** [comment]
Woah awesome, thanks for trying!! I'll try again in a few minutes and if it works we can finally merge this :tada:

**Comment by mxstbr** [comment]
It works!! :tada: but now `npm run build` no longer works... @7rulnik any ideas?

**Comment by jurekbarth** [comment]
@mxstbr i think webpack changed the flagname to `--color` instead of `--colors`

**Comment by mxstbr** [comment]
@7rulnik can you try the above thing and push it to your branch? Then we can finally merge this!

Thanks so much for the help @jurekbarth!

**Comment by 7rulnik** [comment]
Here we go!

**Comment by mxstbr** [comment]
Awesome!! Can you merge the `v3.0.0` branch into this so we resolve the merge conflicts? Thanks!

**Comment by mxstbr** [comment]
All passed, almost looking good…

**Comment by mxstbr** [comment]
DONE, we are on webpack v2, we are on webpack v2! :tada: thanks @7rulnik!

**Comment by oliverturner** [comment]
Holy smokes! Congrats @7rulnik! 

Heroic work 👍

**Comment by mxstbr** [comment]
God damn now we don't have 99% test coverage anymore 😢

**Pull request merged**